File: 1910-remove-all-occurrences-of-a-substring/main.go
func removeOccurrences(s string, part string) string {
    return helper(s, part)
}

func helper(substr string, part string) string {
    if !strings.Contains(substr, part) {
        return substr
    }

    substr = strings.Replace(substr, part, "", 1)
    substr = helper(substr, part)
    return substr
}

File: 1910-remove-all-occurrences-of-a-substring/main.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        
        def _helper(substr: str) -> str:
            
            if part not in substr:
                return substr

            substr = substr.replace(part, '', 1)
            substr = _helper(substr)
            return substr


        return _helper(s)
